docs/developing.md

```diff
@@ -57,3 +57,47 @@ $ cd <application project directory>
 $ yalc remove @hypothesis/frontend-shared
 $ yarn --check-files
 ```
+
+## Adding new components
+
+Before adding a component, determine what category of component it will be — simple, presentational or composite (details can be found in the Pattern Library) — and what group of components it belongs with (e.g. Layout, Data Display).
+
+Each component in this package should have:
+
+1. A component module in `src/components`
```

Comment:
Perhaps worth indicating the directory structure here, including groups (eg. `src/components/{group}/{ComponentName}.js`.